### PR TITLE
Use ERB for share.html template

### PR DESCRIPTION
**Pull request by jmondo**
gsub was causing invalid json when json contained lots of single quotes

### CODE REVIEW

**Comment by jmondo** [comment]
before 
screen shot 2016-10-14 at 3 31 41 pm

**Comment by jmondo** [comment]
after
```
 {
   "execute_type":3,
   "formatted_command_string":"INSERT INTO `user_events` (`user_id`, `organization_id`, `verb`, `object`, `product`, `referrer`, `items_count`, `status_code`, `params`, `properties`, `created_at`, `updated_at`, `event_at`, `git_commit`) VALUES (165, 1, 'show', 'tests', 'search', '', 1, 200, '{\\\"id\\\":\\\"111\\\"}', '{\\\"user_agent\\\":\\\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.143 Safari/537.36\\\"}', '2016-10-14 21:53:21', '2016-10-14 21:53:21', '2016-10-14 21:53:21', '5aa933e')",
   "stack_trace_snippet":"",
   "start_milliseconds":553.7,
   "duration_milliseconds":0.3,
   "first_fetch_duration_milliseconds":0.3,
   "parameters":null,
   "parent_timing_id":"1a71ivwvw4wtkaw1xf2q",
   "is_duplicate":false
}
```

File: lib/mini_profiler/profiler.rb [hunk 1]
```diff
@@ -109,14 +109,15 @@ def serve_results(env)
...
-        html.sub!('{duration}', page_struct.duration_ms.round(1).to_s)
-        html
+      path = "#{env['RACK_MINI_PROFILER_ORIGINAL_SCRIPT_NAME']}#{@config.base_url_path}"
+      version = MiniProfiler::ASSET_VERSION
+      json = result_json
+      includes = get_profile_script(env)
+      name = page_struct[:name]
+      duration = page_struct.duration_ms.round(1).to_s
+
+      template = MiniProfiler.share_template.dup
```
Comment (SamSaffron) [inline lib/mini_profiler/profiler.rb]: hmmm is the dup there needed?

Comment (jmondo) [inline lib/mini_profiler/profiler.rb]: nope!

**Comment by jmondo** [comment]
removed dup

File: lib/mini_profiler/profiler.rb [hunk 2]
```diff
@@ -109,14 +109,15 @@ def serve_results(env)
...
-        html
+      path = "#{env['RACK_MINI_PROFILER_ORIGINAL_SCRIPT_NAME']}#{@config.base_url_path}"
+      version = MiniProfiler::ASSET_VERSION
+      json = result_json
+      includes = get_profile_script(env)
+      name = page_struct[:name]
+      duration = page_struct.duration_ms.round(1).to_s
+
+      template = MiniProfiler.share_template
+      ERB.new(template).result(binding)
```
Comment (SamSaffron) [inline lib/mini_profiler/profiler.rb]: one more thing, what if we memoize ERB.new(template), then we get to reuse it and cut down on parse time no?

Comment (jmondo) [inline lib/mini_profiler/profiler.rb]: ah yes, will do

**Comment by SamSaffron** [comment]
perfect! 

👍

**Pull request merged**